### DISCUSSION TITLE
Is it possible to have Ethernet and WiFi as network interface enabled at a time ?

### DISCUSSION THREAD

**Discussion by enthunilu** [zephyrproject-rtos/zephyr#49417] [Q&A] [answered]
Hello All,

I am just trying to understand possibility of two network interfaces availability in a system at a time.
One as Ethernet and another one as Wi-Fi.

Can anyone share any input or some pointers on this ?

Thanks!!

**Comment by mohanrajd-eaton** [comment]
I am also having same question, is it possible to enable two network interface at a time. Actually, we have a requirement to enable both RNDIS and WIFI together. we have tried integrating RNDIS with WIFI sample application, but it is always taking WIFI as a network interface and RNDIS is not at all working with WIFI. Can anyone guide me on this

**Comment by mohanrajd-eaton** [comment]
Adding more information regarding the issue

we are using U5 board and muRata 1Dx as WIFI interface for our testing. we are trying to test RNDIS functionality in WiFi sample code by adding required project configuration for RNDIS. After adding project configuration, we have tested. but it is always taking WiFi as network interface and it is not taking RNDIS. Even first time we are getting exception as shown below when we connect USB cable.

First time getting exception as shown below:
![image](https://user-images.githubusercontent.com/111718795/229020686-cb433b0c-b10e-4099-9ea6-716e81376bf2.png)

Getting connected to WiFI in second attempt(After restart due to exception):
![image](https://user-images.githubusercontent.com/111718795/229021599-34ac7594-e14e-4b63-9bc6-d10f8ffa378f.png)

WIFI and RNDIS working fine individually, but it is not working together.

Below are configuration added to test RNDIS along with WIFI
![image](https://user-images.githubusercontent.com/111718795/229021131-c6a5888b-1040-482c-9f75-114d9f606ed8.png)

**Comment by jukkar** [comment]
@mohanrajd-eaton Both the Wifi and USB Ethernet should be possible to be enabled at the same time, the network stack does not restrict the interface types that are active simultaneously. Perhaps there is some limit that is hit as having multiple interfaces requires more resources from the system. For the stack trace, you could try to increase the net_mgmt stack size (default is 768 bytes), try to adjust the `CONFIG_NET_MGMT_EVENT_STACK_SIZE` to higher value.

**Reply by mohanrajd-eaton** [reply]
@jukkar Thanks for your input will try by increasing CONFIG_NET_MGMT_EVENT_STACK_SIZE to higher value and let you know.

**Reply by mohanrajd-eaton** [reply, marked as answer]
@jukkar After increasing size of CONFIG_NET_MGMT_EVENT_STACK_SIZE, Ethernet and WiFi working fine. Thanks million for your input.
